src/iterators/mod.rs

```diff
@@ -142,7 +142,9 @@ impl<'a, T: 'a, P> Iterator for HistogramIterator<'a, T, P>
                 return None;
             }
 
-            // have we yielded all non-zeros in the histogram?
+            // TODO should check if we've reached max, not count, to avoid early termination
+            // on histograms with very large counts whose total would exceed u64::max_value()
```

Comment:
Yeah, that's a good point.

Reply:
Though doesn't `running_total` have the same issue?

Reply:
It absolutely does, but we could limit the damage to only saturating `total_count_to_index` and still continuing to iterate until we've reached the max. I've got another branch I'm working on that does this.